nx-X11/programs/Xserver/hw/nxagent/Events.c

```diff
@@ -3570,6 +3571,12 @@ int nxagentHandleReparentNotify(XEvent* X)
         result = XQueryTree(nxagentDisplay, DefaultRootWindow(nxagentDisplay),
                                 &root_return, &parent_return, &children_return, &nchildren_return);
 
+        if (children_return)
+        {
+          XFree(children_return);
+          children_return = NULL;
+        }
+
```

Comment:
I think this is wrong too, children_return is free'd here but used proably used in line 3582

Reply:
right